assets/prototype/domain/shared/predicates/shared/selectionPredicates/index.test.ts

```diff
@@ -0,0 +1,143 @@
+/**
+ * External dependencies
+ */
+import { omit } from 'ramda';
+
+/**
+ * Internal dependencies
+ */
+import {
+	entityDbId,
+	entityGuId,
+	entityHasDbId,
+	entityHasGuid,
+	findEntityByDbId,
+	findEntityByGuid,
+	entitiesWithDbIdInArray,
+	entitiesWithGuIdInArray,
+} from './index';
+import { nodes as datetimes } from '../../../../../domain/eventEditor/data/queries/datetimes/test/data';
+import { nodes as tickets } from '../../../../../domain/eventEditor/data/queries/tickets/test/data';
+
+describe('entityDbId', () => {
+	it('should return dbId for each entity', () => {
+		datetimes.forEach((entity) => {
+			expect(entityDbId(entity)).toBe(entity.dbId);
+		});
+	});
+});
+
+describe('entityGuId', () => {
+	it('should return id for each entity', () => {
+		datetimes.forEach((entity) => {
+			expect(entityGuId(entity)).toBe(entity.id);
+		});
+	});
+});
+
+describe('entityHasDbId', () => {
+	it('should return true if entity has dbId', () => {
+		datetimes.forEach((entity) => {
+			expect(entityHasDbId(entity.dbId)(entity)).toBe(true);
+		});
+	});
+});
+
+describe('entityHasGuid', () => {
+	it('should return true if entity has dbId', () => {
+		datetimes.forEach((entity) => {
+			expect(entityHasGuid(entity.id)(entity)).toBe(true);
+		});
+	});
+});
+
+describe('findEntityByDbId', () => {
+	it('should return the entity with corresponding dbId', () => {
+		const dbIds = [datetimes[0].dbId, datetimes[1].dbId, datetimes[2].dbId];
+
+		dbIds.forEach((dbId) => {
+			const entity = findEntityByDbId(datetimes)(dbId);
+			expect(entity.dbId).toBe(dbId);
+		});
+	});
+
+	it('should return undefined if there is no corresponding entity to the specified dbId', () => {
+		const dbId = 101001110011110001110011;
```

Comment:
Can JS even handle that integer? 😄

Reply:
In ES2020 they added `BigInt`

Reply:
And also in TS 3.2 `BigInt` has been added: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-2.html#bigint

Reply:
uh oh... Vadim speaks binary... going to have to give him the Turing Test!